src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationSection.cs

```diff
@@ -30,6 +30,14 @@ public ConfigurationSection(IConfigurationRoot root, string path)
             _path = path;
         }
 
+        /// <summary>
+        /// Initializes a new instance.
+        /// </summary>
+        /// <param name="root">The configuration root.</param>
+        /// <param name="path">The path to this section.</param>
+        public ConfigurationSection(ConfigurationRoot root, string path) : this((IConfigurationRoot)root, path)
```

Comment:
```suggestion
        [Obsolete]
        public ConfigurationSection(ConfigurationRoot root, string path) : this((IConfigurationRoot)root, path)
```

Reply:
We cannot add obsolete here since it would be source-breaking.  The compiler would prefer the concrete type over interface overload.  I don't think there is any harm in having this overload.

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANmgExAGoAfAAQCYBGAWACgGA3AQygAIx2BedgOxgB3dgGEAFAEoA3A2Zt2wHvyHsAQmLDTZjeuQDM7KqPYh2ASQBiLMDAYBvBgF9t+9tj4YYUAGbWY5qxt2O2cdVyNVewZ2GMMDdUs/dl8bCWDQ2PYAbQB5YABnCFwYTwBdaNjXdRFONNMMAAtsfLExRNTNdO1HIA===